lib/libzpool/os/linux/compat.c

```diff
@@ -0,0 +1,30 @@
+/*
```

Comment:
I think you mistakenly added this to the wrong directory (since you added it to the right header).  If we instead add this to `lib/libzutil/os/linux/zutil_compat.h` then we'll be able to use it from both `libzfs` and `libzpool`.